### PR TITLE
feat: Add trait and features setting for enabling or disabling hxl 

### PR DESCRIPTION
**Pull request by rowasc**
{WIP}feat: Add trait and features setting for enabling or disabling hxl in api #2815 
This pull request makes the following changes:
- Adds feature flag
- Add Features facade
- Add Features class
- Add demo hxl route to show feature toggle working

TODO: 
- Goto to /v3/api/hxl -> should 404
- Enabled HXL feature. 
- Goto to /v3/api/hxl -> returns junk JSON

Test checklist:
- [ ] Checklist in #2815 

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform#2815

Ping @ushahidi/platform

### CODE REVIEW

**Comment by rowasc** [comment]
@rjmackay opening so we work on this

File: src/App/Init.php [hunk 1]
```diff
@@ -815,6 +815,12 @@
     return array_filter($config['data-providers']);
 });
 
+// HDX feature config
+$di->set('hxl.enabled', function () use ($di) {
```
Comment (rjmackay) [inline src/App/Init.php]: I really dislike this pattern. It makes feature flag way too complicated.
I'd prefer a more laravel like approach i.e. define a `Feature` facade and a `feature` helper. So you simple call `Feature::isEnabled('hxl')`.
TL;DR: Don't bother with injecting this

File: app/Tools/Features.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	{
+		$this->config = $config;
+	}
+
+	/**
+	 * Check if a feature is enabled
+	 * @param  string  $feature
+	 * @return boolean
+	 */
+	public function isEnabled($feature)
```
Comment (rjmackay) [inline app/Tools/Features.php]: It might be good to add a method we can call directly in our validators too. Future music though.

File: app/Tools/Features.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+class Features
+{
+
+	/**
+	 * @param array $config
+	 */
+	public function __construct(array $config)
+	{
+		$this->config = $config;
```
Comment (rjmackay) [inline app/Tools/Features.php]: Right now this is direct from config. Future plans:
1. Handle defaults in the class, not config
2. Add some sort of environment based config so you can turn things on in dev/staging/prod easily, without db wrangling

File: routes/web.php [hunk 4]
```diff
@@ -240,6 +240,10 @@
         });
     });
 
+    if (Features::isEnabled('hxl')) {
```
Comment (rjmackay) [inline routes/web.php]: @rowasc added this to demo how we could toggle entire routes based on a feature

Comment (rowasc) [reply routes/web.php]: yay thank you

**Comment by rowasc** [comment]
@rjmackay the TODOs seem to be done already , should we merge?
